src/main.zig

```diff
@@ -1125,7 +1126,12 @@ fn buildOutputType(
                         try clang_argv.appendSlice(it.other_args);
                     },
                     .positional => {
-                        const file_ext = Compilation.classifyFileExt(mem.spanZ(it.only_arg));
+                        var file_ext: Compilation.FileExt = undefined;
+                        if (treat_as_file_ext == .unknown) {
+                            file_ext = Compilation.classifyFileExt(mem.spanZ(it.only_arg));
+                        } else {
+                            file_ext = treat_as_file_ext;
```

Comment:
Is this correct in the presence of both object and source inputs? I think you only want the option to influence the latter.

Reply:
@LemonBoy, according to https://clang.llvm.org/docs/ClangCommandLineReference.html :
```
-x<language>, --language <arg>, --language=<arg>¶
Treat subsequent input files as having type <language>
```

... so I would think you're not supposed to specify any other kind of inputs after a -x / --language option?

Reply:
That was a question for you. I've just checked and clang is blindly treating any input (even binary ones) as source code, this logic is correct.